### PR TITLE
pipeline: keep shard files isolated when merging

### PR DESCRIPTION
**Pull request by adamdecaf**

### CODE REVIEW

File: internal/pipeline/merging.go [hunk 1]
```diff
@@ -120,12 +117,14 @@ func (m *filesystemMerging) HandleCancel(cancel incoming.CancelACHFile) error {
...
-	if err := os.Rename(m.baseDir, newdir); err != nil {
+	olddir := filepath.Join(m.baseDir, m.shard.Name)
+
+	newdir := filepath.Join(filepath.Dir(m.baseDir), fmt.Sprintf("%s-%v", m.shard.Name, time.Now().Format("20060102-150405")))
+
+	if err := os.Rename(olddir, newdir); err != nil {
 		return newdir, err
 	}
-	return newdir, os.Mkdir(m.baseDir, 0777) // create m.baseDir again
+	return newdir, nil // os.Mkdir(m.baseDir, 0777) // create m.baseDir again
```
Comment (darwinz) [inline internal/pipeline/merging.go:127]: The comment doesn't seem relevant now, should we delete it?

**Review by darwinz — APPROVED** [review]

**Review by darwinz — APPROVED** [review]

File: internal/pipeline/merging.go [hunk 2]
```diff
@@ -119,13 +121,15 @@ func (m *filesystemMerging) HandleCancel(cancel incoming.CancelACHFile) error {
 }
 
 func (m *filesystemMerging) isolateMergableDir() (string, error) {
-	// rename m.baseDir so we're the only accessor for it, then recreate m.baseDir
-	parent, _ := filepath.Split(m.baseDir)
-	newdir := filepath.Join(parent, time.Now().Format("20060102-150405"))
-	if err := os.Rename(m.baseDir, newdir); err != nil {
+	// rename the shard directory so we're the only accessor for it, then recreate it
```
Comment (alovak) [inline internal/pipeline/merging.go:124]: where do we recreate the directory?

Comment (alovak) [reply internal/pipeline/merging.go:124]: it seems we create the directory if it does not exist when we handle the received file. am I right?

Comment (adamdecaf) [reply internal/pipeline/merging.go:124]: In the same location it was previously? This is isolating `storage/mergable/` and recreating it.

Comment (alovak) [reply internal/pipeline/merging.go:124]: I see that previously it was `return newdir, os.Mkdir(m.baseDir, 0777)` - we were re-creataing dir on the return, while now I don't see it.

but I see we have `os.MkdirAll(path, 0777)` inside `writeACHFile`.

Comment (adamdecaf) [reply internal/pipeline/merging.go:124]: Yea. I had to make [this change as well](https://github.com/moov-io/achgateway/commit/9312072424c975e11005d7608de734c05b53150a) so directories are created as needed.

**Pull request merged**